.github/workflows/build.yml

```diff
@@ -0,0 +1,41 @@
+name: Build and Test
+
+on:
+  push: # any branch
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: make -C tests EXTRA_CFLAGS="-W -Wall -Wextra -Wswitch-default"
+    - run: make -C tests clean ; make -C tests pedantic
+    - run: make -C tests clean ; make -C tests pedantic EXTRA_CFLAGS=-DNO_DECLTYPE
+    - run: make -C tests clean ; make -C tests cplusplus
+    - run: make -C tests clean ; make -C tests cplusplus EXTRA_CFLAGS=-DNO_DECLTYPE
+  build-clang:
+    runs-on: ubuntu-latest
+    env:
+      CC: clang
+      CXX: clang++
+    steps:
+    - uses: actions/checkout@v3
+    - run: make -C tests EXTRA_CFLAGS="-W -Wall -Wextra -Wswitch-default"
+    - run: make -C tests clean ; make -C tests pedantic
+    - run: make -C tests clean ; make -C tests pedantic EXTRA_CFLAGS=-DNO_DECLTYPE
+    - run: make -C tests clean ; make -C tests cplusplus
+    - run: make -C tests clean ; make -C tests cplusplus EXTRA_CFLAGS=-DNO_DECLTYPE
+  build-mac:
+    runs-on: macos-latest
+    needs:
+      - build
+      - build-clang
```

Comment:
I've set the MacOS CI to only run if the Linux CIs pass, since even though GitHub Actions is free for open-source repos, there might be hidden soft caps (or maybe they'll introduce caps in the future).

Normally MacOS CI costs 10x the Linux CIs on GitHub Actions.

Reply:
That makes sense. Or maybe gate it only on `build-clang`, since Mac uses Clang. (But I believe I see how to make that change myself — just delete one line, right? :))

Naming nit: let's rename `build` to `build-gcc`, if I understand it correctly.

Reply:
Renamed to `build-gcc` and `build-clang` in 5ddf4ef3133c2df881044673ae257a46288e7a33

I've also combined the MacOS builds into `build-gcc` and `build-clang`, since even though it's a random API minilanguage, at least this one is pretty basic.

(plus, it's probably overkill to try to reduce our MacOS CI usage. This repo's CI takes only 1-2 minutes, and other projects have hours worth of CI time)

And, the UI is pretty nice if you use the `matrix` feature:

![image](https://user-images.githubusercontent.com/19716675/184752619-3d11d226-f09e-45aa-8fb6-5a3e4bd730db.png)